src/components/Admin/UserPage/AddSensorModal.tsx

```diff
@@ -0,0 +1,369 @@
+        const querySnapshot = await firestore
+          .collection('sensors')
+          .where('purpleAirId', 'in', [purpleAirId, +purpleAirId])
+          .get();
+
+        // If the query snapshot is not empty, a sensor doc with this PurpleAir ID
+        // already exists in Firestore
+        if (querySnapshot.empty) {
+          const purpleAirApiUrl = `https://api.purpleair.com/v1/sensors/${purpleAirId}`;
+          const purpleAirResponse = await axios({
+            method: 'GET',
+            url: purpleAirApiUrl,
+            headers: {
+              'X-API-Key': process.env.REACT_APP_PURPLEAIR_READ_API_KEY,
+            },
+          });
+
+          const sensorData = purpleAirResponse.data.sensor;
+          setLatitude(sensorData.latitude);
+          setLongitude(sensorData.longitude);
+          setSensorName(sensorData.name);
+          setShowConfirmPage(true);
+        } else {
+          setError(t('sensors.sensorAlreadyExists'));
+        }
+      } catch (error) {
+        setError(t('sensors.unableToAddSensor'));
+      } finally {
+        setIsLoading(false);
+      }
+    }
+  }
+
+  /**
+   * Allows a user to go back to the first form page from the confirm sensor page.
+   * @param event - click button event
+   */
+  function goBackToStart(event: MouseEvent): void {
+    event.preventDefault();
+
+    // Modal state
+    setShowHelp(false);
+    setError('');
+    setIsLoading(false);
+    setShowConfirmPage(false);
+    setShowFinalPage(false);
+    setSensorAdded(false);
+
+    // Sensor state
+    setLatitude(0);
+    setLongitude(0);
+    setSensorName('');
+  }
+
+  /**
+   * Adds a new sensor by creating a sensor doc in the sensors collection in
+   * Firestore and adding the sensor to the PurpleAir group used by the backend
+   * @param event - click button event
+   */
+  async function finishAddSensor(event: MouseEvent): Promise<void> {
+    event.preventDefault();
+
+    if (isAdmin) {
+      setIsLoading(true);
+      const purpleAirIdNumber = +purpleAirId;
+      // Add to PurpleAir Group
+      const purpleAirGroupApiUrl =
+        'https://api.purpleair.com/v1/groups/490/members';
+
+      try {
+        // Add the sensor to the sensor group 490
+        await axios({
+          method: 'POST',
+          url: purpleAirGroupApiUrl,
+          headers: {
+            'X-API-Key': process.env.REACT_APP_PURPLEAIR_WRITE_API_KEY,
+          },
+          params: {
+            sensor_index: purpleAirIdNumber, // eslint-disable-line camelcase
+          },
+        });
+
+        // Create a sensor doc for the added sensor
+        await firestore.collection('sensors').add({
+          name: sensorName,
+          purpleAirId: purpleAirIdNumber,
+          latitude: latitude,
+          longitude: longitude,
+          isActive: true,
+          isValid: false,
+          lastValidAqiTime: null,
+          lastSensorReadingTime: null,
```

Comment:
Do we need to start a readings subcollection or does that happen automatically when we populate with the first historical reading

Reply:
This happens automatically